service/history/execution/context.go

```diff
@@ -1215,6 +1217,20 @@ func (c *contextImpl) updateWorkflowExecutionWithRetry(
 		return err
 	}
 
+	//Calculating the TTL for workflow Execution.
+	domainObj, _ := c.shard.GetDomainCache().GetDomainByID(c.domainID)
+	config := domainObj.GetConfig()
+	retention := time.Duration(config.Retention)
+	daysInSeconds := int((retention + ttlBufferDays) * dayToSecondMultiplier)
+
+	startTime := request.UpdateWorkflowMutation.ExecutionInfo.StartTimestamp
+	if time.Time.IsZero(startTime) {
+		request.TTLInSeconds = daysInSeconds
+	} else {
+		request.TTLInSeconds = int(request.UpdateWorkflowMutation.ExecutionInfo.DecisionStartToCloseTimeout) - int(time.Now().Sub(startTime).Seconds()) + daysInSeconds
```

Comment:
I think it should not be `DecisionStartToCloseTimeout`, and it should be the workflow timeout, not the decision timeout

Reply:
Another thing is to make sure `int(request.UpdateWorkflowMutation.ExecutionInfo.DecisionStartToCloseTimeout) - int(time.Now().Sub(startTime).Seconds())` is >= 0. 
It is possible to be negative when there is an update after workflow timeout.